src/main/java/com/google/error_analyzer/backend/Pagination.java

```diff
@@ -0,0 +1,135 @@
+
+
+@WebServlet("/pagination")
+public class Pagination extends HttpServlet {
+
+    private String field = "name";
+    private static String ERROR = "errors";
+    private static final Logger LOG =
+        LogManager.getLogger(Pagination.class);
+    // keep noOfPages a odd no so that there are equal pages in front 
+    // and back 
+    private int noOfPages = 5;
+    private LogDao database = new LogDao();
+    private LogDaoHelper databaseHelper = new LogDaoHelper();
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        int page = Integer.parseInt(request.getParameter("requestedPage"));
+        String fileName = request.getParameter("fileName");
+        String fileType = request.getParameter("fileType");
+        int recordsPerPage =
+            Integer.parseInt(request.getParameter("recordsPerPage"));
+
+        response.setContentType("application/json");
+        if (fileName.length() == 0 || !database.fileExists(fileName)) {
+            String json = returnEmptyObject();
+            response.getWriter().println(json);
+            return;
+        }
+        String json =
+            fetchAndReturnResponse(page, fileName, fileType, recordsPerPage);
+        response.getWriter().println(json);
+    }
+
+    // fetches logs from database and return json for the same
+    private String fetchAndReturnResponse(int page, String fileName,
+        String fileType, int recordsPerPage) {
+        int start = (page - 1) * recordsPerPage;
+        int size = recordsPerPage;
+        if (page == 1) {
+            size = noOfPages * recordsPerPage;
+        }
+        if (fileType.equals(ERROR)) {
+            fileName += "error";
+        }
+        return fetchData(start, size, fileName, fileType);
+    }
+
+    // interact with database using dao functions
+    private String fetchData(int start, int size, String fileName,
+        String fileType) {
+        try {
+            SearchHit[] searchHits =
+                database.getAll(fileName, start, size);
+            ImmutableList < String > hitIds =
+                databaseHelper.hitId(searchHits);
+            ImmutableList < String > hitFieldContent =
+                databaseHelper.hitFieldContent(searchHits, field);
+            if (hitIds == null) {
+                return convertToJson(new ArrayList());
+            }
+            return addFetchResultToData(fileType, hitIds, hitFieldContent);
+        } catch (IOException exception) {
+            LOG.error(exception);
+        }
+        return convertToJson(new ArrayList());
+    }
+
+    // add errorfixes and text highlightes 
+    private String addFetchResultToData(String fileType,
+        ImmutableList < String > hitIds,
+        ImmutableList < String > hitFieldContent) {
+        ArrayList < String > data = new ArrayList();
+        SearchErrors searchErrors = new SearchErrors();
+        HashMap < String, String > search =
```

Comment:
I am not clear how "search" is being used.

Reply:
search contains all the highlighted matches mapped to doc id that we got while applying full text search